packages/protocol/test/governance/double_signing_slasher.ts

```diff
@@ -0,0 +1,217 @@
+  const validatorScoreParameters = {
+    exponent: new BigNumber(5),
+    adjustmentSpeed: toFixed(0.25),
+  }
+  const membershipHistoryLength = new BigNumber(5)
+  const maxGroupSize = new BigNumber(5)
+
+  // A random 64 byte hex string.
+  const blsPublicKey =
+    '0x4fa3f67fc913878b068d1fa1cdddc54913d3bf988dbe5a36a20fa888f20d4894c408a6773f3d7bde11154f2a3076b700d345a42fd25a0e5e83f4db5586ac7979ac2053cd95d8f2efd3e959571ceccaa743e02cf4be3f5d7aaddb0b06fc9aff00'
+  const blsPoP =
+    '0xcdb77255037eb68897cd487fdd85388cbda448f617f874449d4b11588b0b7ad8ddc20d9bb450b513bb35664ea3923900'
+  const commission = toFixed(1 / 100)
+
+  const registerValidator = async (address: string) => {
+    await mockLockedGold.setAccountTotalLockedGold(address, validatorLockedGoldRequirements.value)
+    const publicKey = await addressToPublicKey(address, web3.eth.sign)
+    await validators.registerValidator(
+      // @ts-ignore bytes type
+      publicKey,
+      // @ts-ignore bytes type
+      blsPublicKey,
+      // @ts-ignore bytes type
+      blsPoP,
+      { from: address }
+    )
+  }
+
+  const registerValidatorGroup = async (group: string) => {
+    await mockLockedGold.setAccountTotalLockedGold(group, groupLockedGoldRequirements.value)
+    await validators.registerValidatorGroup(commission, { from: group })
+  }
+
+  const registerValidatorGroupWithMembers = async (group: string, members: string[]) => {
+    await registerValidatorGroup(group)
+    for (const address of members) {
+      await registerValidator(address)
+      await validators.affiliate(group, { from: address })
+      if (address === members[0]) {
+        await validators.addFirstMember(address, NULL_ADDRESS, NULL_ADDRESS, { from: group })
+      } else {
+        await validators.addMember(address, { from: group })
+      }
+    }
+  }
+
+  const nonOwner = accounts[1]
+  const validator = accounts[1]
+
+  beforeEach(async () => {
+    accountsInstance = await Accounts.new()
+    // Do not register an account for the last address so it can be used as an authorized validator signer.
+    await Promise.all(
+      accounts.slice(0, -1).map((account) => accountsInstance.createAccount({ from: account }))
+    )
+    mockElection = await MockElection.new()
+    mockLockedGold = await MockLockedGold.new()
+    registry = await Registry.new()
+    validators = await Validators.new()
+    slasher = await DoubleSigningSlasher.new()
+    await accountsInstance.initialize(registry.address)
+    await registry.setAddressFor(CeloContractName.Accounts, accountsInstance.address)
+    await registry.setAddressFor(CeloContractName.Election, mockElection.address)
+    await registry.setAddressFor(CeloContractName.LockedGold, mockLockedGold.address)
+    await registry.setAddressFor(CeloContractName.Validators, validators.address)
+    await registry.setAddressFor(CeloContractName.DoubleSigningSlasher, slasher.address)
+    await registry.setAddressFor(CeloContractName.DowntimeSlasher, accounts[5])
```

Comment:
Do we need to register all of these contracts? Same question for other tests

Reply:
Perhaps some of them can be dropped if `MockValidators` was used.